src/java/org/apache/cassandra/index/sai/disk/vector/VectorCompression.java

```diff
@@ -18,42 +18,64 @@
 
 package org.apache.cassandra.index.sai.disk.vector;
 
-import io.github.jbellis.jvector.pq.BQVectors;
-import io.github.jbellis.jvector.pq.CompressedVectors;
-import io.github.jbellis.jvector.pq.PQVectors;
+import java.util.Objects;
 
 public class VectorCompression
 {
+    public static final VectorCompression NO_COMPRESSION = new VectorCompression(CompressionType.NONE, -1, -1);
```

Comment:
Sometimes it is easy to know the original size when creating a VectorCompression with type NONE, but not always.  So we make it official that this is optional for NONE, and check in the accessors to make sure it's not getting used accidentally.